### DISCUSSION TITLE
The design behind naming of S3StageSource

### DISCUSSION THREAD

**Discussion by Xuanwo** [databendlabs/databend#5250] [Q&A] [answered]
in `common/planners/src/plan_read_datasource.rs`, we have `SourceInfo` like:

```rust
#[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq)]
pub enum SourceInfo {
    // Normal table source, 'fuse/system'.
    TableSource(TableInfo),

    // S3 internal/external source, 's3://'.
    S3StageSource(S3StageTableInfo),
}
```

And in `S3StageTableInfo` we will have:

```
S3StageTableInfo -> UserStageInfo -> StageType（Internal/External）
                                  -> StageParams -> StageStorage -> S3  
```

---

Is it better to rename `S3StageTableInfo` into `StageTableInfo`? So that we can add `azblob`/`hdfs` as stage storage inside?

Please correct me if I misunderstand anything.

Especially ping @sundy-li for comments.

**Comment by sundy-li** [comment, marked as answer]
Yes, I think it's reasonable.
